src/DBus/PermissionStore.vala

```diff
@@ -0,0 +1,20 @@
+/*
+ * SPDX-License-Identifier: LGPL-3.0-or-later
+ * SPDX-FileCopyrightText: 2023 elementary, Inc. (https://elementary.io)
+ */
+
+[DBus (name = "org.freedesktop.impl.portal.PermissionStore")]
+interface SecurityPrivacy.PermissionStore : Object {
+    public abstract void delete_permission (string table, string id, string app) throws Error;
+    public abstract void delete (string table, string id) throws Error;
+    public abstract void list (string table, out string[] ids) throws Error;
+    public abstract void set_permission (string table, bool create, string id, string app, string[] permissions) throws Error;
+    public abstract void set_value (string table, bool create, string id, Variant data) throws Error;
+    public abstract void lookup (string table, string id, out HashTable<string, Variant> permissions, out Variant data) throws Error;
```

Comment:
I used [vala-dbus-binding-tool](https://github.com/freesmartphone/vala-dbus-binding-tool) with [org.freedesktop.impl.portal.PermissionStore.xml](https://github.com/flatpak/xdg-desktop-portal/blob/main/data/org.freedesktop.impl.portal.PermissionStore.xml):
```vala
[DBus (name = "org.freedesktop.impl.portal.PermissionStore", timeout = 120000)]
public interface PermissionStore : GLib.Object {

	[DBus (name = "version")]
	public abstract uint version {  get; }

	[DBus (name = "Lookup")]
	public abstract void lookup(string table, string id, out GLib.HashTable<string, string[]> permissions, out GLib.Variant data) throws DBusError, IOError;

	[DBus (name = "Set")]
	public abstract void set(string table, bool create, string id, GLib.HashTable<string, string[]> app_permissions, GLib.Variant data) throws DBusError, IOError;

	[DBus (name = "Delete")]
	public abstract void delete(string table, string id) throws DBusError, IOError;

	[DBus (name = "SetValue")]
	public abstract void set_value(string table, bool create, string id, GLib.Variant data) throws DBusError, IOError;

	[DBus (name = "SetPermission")]
	public abstract void set_permission(string table, bool create, string id, string app, string[] permissions) throws DBusError, IOError;

	[DBus (name = "DeletePermission")]
	public abstract void delete_permission(string table, string id, string app) throws DBusError, IOError;

	[DBus (name = "GetPermission")]
	public abstract string[] get_permission(string table, string id, string app) throws DBusError, IOError;

	[DBus (name = "List")]
	public abstract string[] list(string table) throws DBusError, IOError;

	[DBus (name = "Changed")]
	public signal void changed(string table, string id, bool deleted, GLib.Variant data, GLib.HashTable<string, string[]> permissions);
}
```
this can be cleaned-up/aligned with our conventions but it should work :slightly_smiling_face:

Reply:
I was messing with this branch last night and the `a{sas}` variants are an issue as `GLib.HashTable<string, string[]>` won't compile.

```
Compilation failed: 3 error(s), 0 warning(s)
<source>:8.82-8.89: error: Arrays are not supported as generic type arguments
    8 | 	public abstract void lookup(string table, string id, out GLib.HashTable<string, string[]> permissions, out GLib.Variant data) throws DBusError, IOError;
      | 
```

I have an online compilation of this interface here:
https://godbolt.org/z/9Tah45PsY

Reply:
So then your best luck is:
```vala
[DBus (name = "org.freedesktop.impl.portal.PermissionStore", timeout = 120000)]
public interface PermissionStore : GLib.Object {

	[DBus (name = "version")]
	public abstract uint version {  get; }

	[DBus (name = "Lookup")]
	public abstract void lookup(string table, string id, [DBus (signature = "a{sas}")] out GLib.Variant permissions, out GLib.Variant data) throws DBusError, IOError;

	[DBus (name = "Set")]
	public abstract void set(string table, bool create, string id, [DBus (signature = "a{sas}")] GLib.Variant app_permissions, GLib.Variant data) throws DBusError, IOError;

	[DBus (name = "Delete")]
	public abstract void delete(string table, string id) throws DBusError, IOError;

	[DBus (name = "SetValue")]
	public abstract void set_value(string table, bool create, string id, GLib.Variant data) throws DBusError, IOError;

	[DBus (name = "SetPermission")]
	public abstract void set_permission(string table, bool create, string id, string app, string[] permissions) throws DBusError, IOError;

	[DBus (name = "DeletePermission")]
	public abstract void delete_permission(string table, string id, string app) throws DBusError, IOError;

	[DBus (name = "GetPermission")]
	public abstract string[] get_permission(string table, string id, string app) throws DBusError, IOError;

	[DBus (name = "List")]
	public abstract string[] list(string table) throws DBusError, IOError;

	[DBus (name = "Changed")]
	public signal void changed(string table, string id, bool deleted, GLib.Variant data, [DBus (signature = "a{sas}")] GLib.Variant permissions);
}
```
and then do:
```vala
var iter = permissions.iterator ();
string app_id;
string[] app_permissions;
while (iter.next ("{sas}", out app_id, out app_permissions)) {
    //TODO:
}
```